### ISSUE TITLE
Vite projects do not work with npm 7+

### ISSUE THREAD

**Issue by Artur-** [vaadin/flow#12392]
### Description of the bug / feature
```
npm ERR! While resolving: no-name@0.9.4
npm ERR! Found: vite@2.7.0-beta.5
npm ERR! node_modules/vite
npm ERR!   dev vite@"v2.7.0-beta.5" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer vite@"^2.0.0" from vite-plugin-checker@0.3.4
npm ERR! node_modules/vite-plugin-checker
npm ERR!   dev vite-plugin-checker@"0.3.4" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

For now the workaround is to run `npm install -f` manually

This will be resolved at least when Vite 2.7.0 stable is out

**Comment by joheriks** [comment]
Note issue is fixed temporarily by #12397, but keeping ticket open until stable release of Vite is used. The code that adds the `--legacy-peer-deps` parameter to npm should then be removed and this ticket closed.

**Comment by Artur-** [comment]
The code was removed in #12522
